### PR TITLE
provide possibility to set up data dir(Rebased to newest version)

### PR DESCRIPTION
**Pull request by pzhong-livongo**
Pull #262 previously did most of the ground work, but this is now rebased to the newest version of sbt-scoverage.

### CODE REVIEW

**Comment by pzhong-livongo** [comment]
Would appreciate any feedback :) @ckipp01

File: src/main/scala/scoverage/ScoverageKeys.scala [hunk 1]
```diff
@@ -18,6 +18,7 @@ object ScoverageKeys {
   lazy val coverageOutputDebug = settingKey[Boolean]("turn on the debug report")
   lazy val coverageOutputTeamCity = settingKey[Boolean]("turn on teamcity reporting")
   lazy val coverageScalacPluginVersion = settingKey[String]("version of scalac-scoverage-plugin to use")
+  lazy val coverageDataDir = settingKey[File]("directory where the measurements will be stored to use")
```
Comment (ckipp01) [inline src/main/scala/scoverage/ScoverageKeys.scala:21]:
```suggestion
  lazy val coverageDataDir = settingKey[File]("directory where the measurements and report files will be stored")
```

File: src/main/scala/scoverage/ScoverageSbtPlugin.scala [hunk 2]
```diff
@@ -56,11 +56,12 @@ object ScoverageSbtPlugin extends AutoPlugin {
     addCommandAlias("coverageOn", ";set ThisBuild / coverageEnabled := true") ++
     addCommandAlias("coverageOff", ";set ThisBuild / coverageEnabled := false")
 
-  override def projectSettings: Seq[Setting[_]] = Seq(
+  override def projectSettings: Seq[Setting[_]] = super.projectSettings ++ Seq(
```
Comment (ckipp01) [inline src/main/scala/scoverage/ScoverageSbtPlugin.scala:59]: So I get that this is getting the current `projectSettings` and then appending, but I don't often seen this done is sbt plugins. What is the main reason for this change?

**Review by ckipp01 — CHANGES_REQUESTED** [review]
Thanks a lot for this @pzhong-livongo! Just a quick suggestion on the description and a question on super.

File: src/main/scala/scoverage/ScoverageSbtPlugin.scala (hunk 2 above)
Comment (pzhong-livongo) [reply src/main/scala/scoverage/ScoverageSbtPlugin.scala:59]: Now that I think about it I don't think it's necessary and I will remove it.

**Review by ckipp01 — APPROVED** [review]
Perfect. Thanks @pzhong-livongo!

**Comment by pzhong-livongo** [comment]
@ckipp01  Thank you so much for merging this. Would it be possible for a new version to be published?

**Comment by ckipp01** [comment]
> @ckipp01 Thank you so much for merging this. Would it be possible for a new version to be published?

Scala 2.12.15 should be out [any day](https://github.com/scala/scala-dev/issues/783), so as soon as that's released I'll cut a new version of the scalac-plugin and then this as well. So if you can wait (hopefully a week or so), then this will be released along with that.

**Pull request merged**